docs/dg/dev/architecture/dependency-injection.md

```diff
@@ -0,0 +1,215 @@
+
+                continue;
+            }
+
+            $services->load($namespace, $path);
+        }
+    }
+};
+```
+
+### Autowiring
+
+You can use autowiring to automatically inject dependencies into your services. For details, see [Autowiring](https://symfony.com/doc/current/service_container/autowiring.html) in the Symfony documentation.
+
+Since Spryker's core modules are not yet using the Symfony Dependency Injection component, you can't autowire classes from the core. You need to manually add them to your service configuration if you want to inject core dependencies.
+
+## Compiling the container
+
+To compile the container, you have two options:
+
+- On every request to the application, the Kernel checks if a compiled container already exists. If not, or if the cache is outdated, it generates a new one. (perfect in dev mode)
+- Run the following command to build the cache. The application then immediately uses the freshly compiled container.
+
+```bash
+console container:build
+```
+
+## How Dependency Injection is triggered
+
+Dependency injection is activated when a service is first requested from the container. For instance, if a controller is defined as a service in the container, the container instantiates it. If this controller's constructor has arguments (dependencies), the container automatically injects the required services. This process continues down the dependency chain.
+
+If a class is not defined as a service in the container, it will be instantiated the classic Spryker way, and no dependencies will be injected.
+
+## Using services
+
+Once the container is compiled, you can use your services in your application. You can inject them into your controllers, and other services via constructor injection.
+
+### Controller example
+
+Here is an example of how to inject a service into a controller.
+
+First, define your controller and inject the service using constructor property promotion:
+
+`src/Pyz/Yves/MyModule/Controller/IndexController.php`
+
+```php
+<?php
+
+namespace Pyz\Yves\MyModule\Controller;
+
+use Spryker\Yves\Kernel\Controller\AbstractController;
+use Pyz\Yves\MyModule\MyService;
+
+class IndexController extends AbstractController
+{
+    public function __construct(private MyService $myService)
+    {
+    }
+
+    public function indexAction()
+    {
+        $data = $this->myService->getSomeData();
+
+        return $this->view(
+            '@MyModule/views/index/index.twig',
+            ['data' => $data]
+        );
+    }
+}
+```
+
+Because we configured service discovery for the `Pyz\Yves\MyModule` namespace in `config/ApplicationServices.php` and autowiring is enabled, all controllers will be available in the Container when requested.
+
+
+### Model/Service example
+
+Here is an example of a service with a dependency on a repository.
+
+`src/Pyz/MyModule/src/Pyz/MyModule/SomeService.php`
```

Comment:
This structure is not used on the projects.

Reply:
yes, example https://github.com/spryker-shop/b2b-demo-marketplace/tree/master/src/Pyz/Glue